### PR TITLE
feat(resolver.js)Support handleConnection for root query

### PR DESCRIPTION
**Pull request by hank-cp**
Support connection on root Query
```graphql
type Query {
  users(
    first: Int
    after: String
    last: Int
    before: String
  ): UserConnection
}
```
or 
```js
    var userConnection = connectionDefinitions({name: 'User', nodeType: userType});

    schema = new GraphQLSchema({
      query: new GraphQLObjectType({
        name: 'RootQueryType',
        fields: {
          user: {
            type: userConnection.connectionType,
            args: connectionArgs,
            resolve: resolver(User)
          },
          node: nodeField
        }
      })
    })
```
Unit test included :)

### CODE REVIEW

File: src/resolver.js [hunk 1]
```diff
@@ -95,7 +95,15 @@ function resolverFactory(targetMaybeThunk, options = {}) {
...
 
-      return model[list ? 'findAll' : 'findOne'](findOptions);
+      if (options.handleConnection && isConnection(info.returnType)) {
+        return model['findAll'](findOptions).then(function (result) {
+          return handleConnection(result, args);
+        });
+
+      } else {
+        return model[list ? 'findAll' : 'findOne'](findOptions);
+      }
```
Comment (mickhansen) [inline src/resolver.js]:
```
return model[(list || options.handleConnection && isConnection(info.returnType)) ? 'findAll' : 'findOne'](findOptions).then(function (result) {
  if (options.handleConnection && isConnection(info.returnType)) return handleConnection(result, args);
  return result;
});
```

File: src/resolver.js [hunk 2]
```diff
@@ -95,7 +95,12 @@ function resolverFactory(targetMaybeThunk, options = {}) {
         }
       }
 
-      return model[list ? 'findAll' : 'findOne'](findOptions);
+      // eslint-disable-next-line max-len,no-extra-parens
```
Comment (mickhansen) [inline src/resolver.js]: hmm build still failing on linting

File: src/resolver.js [hunk 3]
```diff
@@ -76,26 +76,34 @@ function resolverFactory(targetMaybeThunk, options = {}) {
         findOptions.order = [[model.primaryKeyAttribute, 'ASC']];
       }
 
+      function checkConnection() {
```
Comment (mickhansen) [inline src/resolver.js]: i like this approach but defining the functions everytime seems like a waste, move them outside and have them take all args required please.

**Comment by hank-cp** [comment]
Done, 1 more char triggers eslint warning...huuuu

**Comment by hank-cp** [comment]
This build failed looks not caused by my changes

**Pull request closed without merging**